### PR TITLE
Parse crowbarctl upgrade prechecks with ruby

### PR DESCRIPTION
**Pull request by abel-navarro**
This provides a more genereic output parsing than previous proposed solution based on bash.

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -4464,7 +4464,9 @@ function onadmin_prepare_crowbar_upgrade
         # move nodes to upgrade mode
         safely crowbar_api_request POST $crowbar_api /installer/upgrade/prepare.json
     else
-        if crowbarctl upgrade prechecks --format plain | grep "false" ; then
+        if ! crowbarctl upgrade prechecks --format json | rubyjsonparse \ 
+            'exit false if j.keys.any? { |key| j[key]["required"] && !j[key]["passed"]}'
+
```
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: `then` is missing

Comment (abel-navarro) [inline scripts/qa_crowbarsetup.sh]: right, copy and paste error. Fixing that.

**Comment by abel-navarro** [comment]
CI failed because of unrelated problem (tempest test, server_basic_ops).

**Comment by jdsn** [comment]
@abel-navarro was the changed code path run by the gating at all?

**Comment by abel-navarro** [comment]
@jdsn the output from CI shows it's using this PR, so I guess yes, the CI tested this.

**Comment by jdsn** [comment]
@abel-navarro I am not asking if the PR was used in this PR, but rather if the lines of code were used with the gating job used here? If not, please stop the gating job, click rebuild and change the build parameters accordingly so that the code is really computed.

**Comment by abel-navarro** [comment]
@aspiers agree. In fact I could ear your voice and see your face reminding me this exact thing while typing this code.

@jdsn now I understand your point. Rebuilt the job with the upgrade process which should execute that specific code.

**Comment by jdsn** [comment]
@abel-navarro I will retrigger with proper parameters now

**Pull request merged**